client/packages/system/src/Patient/JsonForms/components/BMI.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from 'react';
+import {
+  composePaths,
+  ControlProps,
+  rankWith,
+  uiTypeIs,
+} from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import {
+  DetailInputWithLabelRow,
+  FORM_LABEL_WIDTH,
+  NumUtils,
+} from '@openmsupply-client/common';
+
+export const bmiTester = rankWith(10, uiTypeIs('BMI'));
+
+const round = (value: number) => Math.round(value * 100) / 100;
+
+const UIComponent = (props: ControlProps) => {
+  const { data, handleChange, label, path } = props;
+  const [bmi, setBmi] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (!data) return;
+
+    const height = NumUtils.parseString(data.height);
```

Comment:
yes, it requires having sibling properties of height and weight. Fails silently if they aren't present of can't be parsed though, so I think this is ok.